types/rheostat/index.d.ts

```diff
@@ -45,7 +45,7 @@ export interface Props extends Events {
   progressBar?: React.ReactType;
   snap?: boolean;
   snapPoints?: number[];
-  values?: number[];
+  values?: [number, number];
```

Comment:
Since the algorithm is customizable, this could in fact contain a million numbers, or zero, or one; `number[]` is correct here.

Reply:
Background: back when @SashaBayan and I wrote the first iteration of this package, we looked at this [`prop-types` definition of the `values` prop](https://github.com/airbnb/rheostat/blob/4a6893659e1c0e53a8138fb1c39a36721b355d3a/src/Slider.jsx#L107) and arrived at `number[]`.

Grab bag of questions:
- Does the `length` of the `values` prop stay the same throughout the entire lifecycle of the component?
- Is there a way to discern the `length` of the `values` prop by looking at certain things, like the type signature of an `Algorithm` object, or the initial `length` of the `values` prop passed in?

I ask about algorithms, because all the algorithms that ship with Rheostat appear to deal with one value at a time, but within the component, `values` is clearly expected to be at least some kind of array (due to the `.map` calls) and each element is meant to be numeric (due to the arithmetic operators used on each value).

Reply:
You could perhaps infer it based on the algorithm types, and then that puts an extra generics cost on consumers who provide an algorithm that accepts a different type for values.

Reply:
Some closer reading in the source code leads me to believe that the concerns are divided like this:
- An algorithm deals with a "numeric" type, one value at a time. This is `number` for most cases, but theoretically there's no reason it can't be something else, like a `BigInt`. (I believe due to its use of only built-in operators, the default linear one also works with `BigInt` values without modification.)
- The `values` prop flowing into the `<Rheostat>` component must be an array of "numeric" values. That prop is the only thing that controls how many handles a slider has.

Based on this, we can introduce two type parameters with good default types:
```ts
interface Props<Numeric = number, Values extends Numeric[] = Numeric[]> {
  min?: Numeric
  max?: Numeric
  values?: Values
}
```

This gives us a few scenarios:
- Out of the box, `Numeric = number`, and `Values = number[]`. It would work exactly the same way as it currently does.
- For @george-dotdev's use case, `Numeric = number`, and `Values = [number, number]`.
- Let's say you want a slider with three handles, each of them represented by a `BigInt`. Set `Numeric = BigInt` and `Values = [BigInt, BigInt, BigInt]` to get that.

I'm omitting a lot of the generics bleeding into all downstream interfaces (`Algorithm`, `PublicEvent`, etc), but does this sound like an accurate description of the semantics of `values` and friends?

Reply:
That seems right to me.

As for handling BigInt, there’s no PropType for that yet; I’ll maybe add one to prop-types soon.